### PR TITLE
12 cycle count model

### PR DESCRIPTION
**Pull request by LucasAnderson1984**
Finished cycle count model and updated controller tests.

### CODE REVIEW

File: spec/controllers/cycle_counts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+      it_behaves_like 'an unauthorized access to a resource'
+    end
+  end
+
+  describe 'DELETE destroy' do
+    before do
+      allow(cycle_count).to receive(:destroy) { destroy? }
+      allow(CycleCount).to receive(:find) { cycle_count }
+
+      xhr :delete, :destroy, id: cycle_count.id
```
Comment (josephbridgwaterrowe) [inline spec/controllers/cycle_counts_controller_spec.rb]: I'm curious as to why you're calling this action with `xhr` and not `delete`?

Comment (LucasAnderson1984) [inline spec/controllers/cycle_counts_controller_spec.rb]: Because I copied one of your delete tests instead of writing it myself.  I'll change it.

**Comment by josephbridgwaterrowe** [comment]
@LucasAnderson1984 looks good for rebase, however Code Climate doesn't look very happy.

**Comment by LucasAnderson1984** [comment]
This looks like a factory duplication not necessarily an app failure?

**Comment by josephbridgwaterrowe** [comment]
@LucasAnderson1984 might be a good time to update the factory to prevent that from happening in the future... I normally achieve this through a sequence and just interpolate the number at the end of the string.

**Comment by josephbridgwaterrowe** [comment]
@LucasAnderson1984 looks like its time for a rebase.

**Comment by josephbridgwaterrowe** [comment]
@LucasAnderson1984 Looks good :cake:

**Pull request merged**